modules/core/arrow-test/src/main/kotlin/arrow/test/laws/ApplicativeLaws.kt

```diff
@@ -12,7 +12,7 @@ import io.kotlintest.properties.forAll
 object ApplicativeLaws {
 
   fun <F> laws(A: Applicative<F>, EQ: Eq<Kind<F, Int>>): List<Law> =
-    FunctorLaws.laws(A, EQ) + listOf(
+    FunctorLaws.laws(A, Gen.int().map { A.just(it) }, EQ) + listOf(
```

Comment:
This is not good. This is part of the reason we need these changes in the first place, please just add `GenK<F>` as a parameter for this.

Reply:
Also use `EqK` instead of `Eq<Kind<F, Int>>`

Reply:
ok, i did a first pass replacing Eq with EqK. that was quite some work ;-) 
making changes to the laws API have a ripple effects on the whole test codebase. 

i will replace Gen with GenK in an extra pass.

Reply:
> making changes to the laws API have a ripple effects on the whole test codebase.

Yup, this is actually a good time to do this pr as most things that currently change in arrow change in fx. Otherwise you'd be drowning in merge fixes ;)
This pr is a lot of work, and if you need help with it, or someone else to do some tasks just ask away :)